File: app/src/main/java/com/example/tdm_android/activities/LoginActivity.java
package com.example.tdm_android.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Dialog;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.example.tdm_android.R;
import com.example.tdm_android.constants.Constants;
import com.example.tdm_android.managers.UserManager;
import com.example.tdm_android.models.User;

public class LoginActivity extends AppCompatActivity {

    TextView tvCreateUser;
    EditText etUsername, etPassword;
    Button btnLogin;
    CheckBox checkRememberUser;
    SharedPreferences pref;
    SharedPreferences.Editor editor;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        String strName = getIntent().getStringExtra(Constants.STR_KEY_USERNAME);
        if(strName != null) { //Si se acaba de registrar
            Toast.makeText(this, Constants.REGISTERED_SUCCESSFULLY_MESSAGE + strName, Toast.LENGTH_SHORT).show();
        }

        initializeVariables();

        if (this.pref.getBoolean(Constants.STR_CHECK_REMEMBER_USER, false)) {//Si había elegido recordar usuario
            redirectToFilterActivity();
        }

        tvCreateUser.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                redirectToCreateUser(view);
            }
        });

        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                login(view);
            }
        });

    }

    public void redirectToCreateUser(View view){
        Intent intent = new Intent(this, RegisterActivity.class);
        startActivity(intent);
    }

    public void login(View view){

        String strUsername = etUsername.getText().toString();
        String strPassword = etPassword.getText().toString();

        final Dialog dialog = new Dialog(LoginActivity.this);
        dialog.setContentView(R.layout.custom_dialog);
        dialog.setCancelable(false);

        TextView tvDialogInfo = (TextView) dialog.findViewById(R.id.tvDialogInfo);
        Button btnDialogOk = (Button) dialog.findViewById(R.id.btnDialogOk);

        btnDialogOk.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dialog.dismiss();
            }
        });

        if ( strUsername.isEmpty() || strPassword.isEmpty()){
            tvDialogInfo.setText(Constants.COMPLETE_ALL_FIELD_MESSAGE);
            dialog.show();
        }else if( isTheCredentialsAreValid(strUsername, strPassword) ) {
            saveDataUser(checkRememberUser.isChecked());
            redirectToFilterActivity();
        }else{
            tvDialogInfo.setText(Constants.INVALID_CREDENTIALS_MESSAGE);
            dialog.show();
        }
    }

    public void redirectToFilterActivity(){
        Intent intent = new Intent(this, FilterActivity.class);
        intent.putExtra(Constants.STR_KEY_USERNAME, etUsername.getText().toString());
        startActivity(intent);
    }
    public boolean isTheCredentialsAreValid(String strUsername, String strPassword){
        try {
            User user = UserManager.getInstance(LoginActivity.this).getOneUserByField("username", strUsername);
            if (user != null && user.getPassword().equals(strPassword)) {return true;}
        } catch (Exception e){
            e.printStackTrace();
        }
        return false;
    }

    private void initializeVariables() {
        tvCreateUser = findViewById(R.id.createUser);
        btnLogin = findViewById(R.id.login);
        etUsername = findViewById(R.id.txtNameUser);
        etPassword = findViewById(R.id.txtPassword);
        checkRememberUser = findViewById(R.id.checkRememberUser);
        pref = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
        editor = pref.edit();
    }

    public void saveDataUser(boolean isRememberUser){
        editor.putString(Constants.STR_USERNAME, etUsername.getText().toString());
        editor.putString(Constants.STR_PASSWORD, etPassword.getText().toString());
        editor.putBoolean(Constants.STR_CHECK_REMEMBER_USER, isRememberUser);
        editor.apply();
    }

}

File: app/src/main/java/com/example/tdm_android/activities/LoginActivity.kt
package com.example.tdm_android.activities

import android.app.Dialog
import com.example.tdm_android.R
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.Intent
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.view.View
import android.widget.*
import androidx.core.content.ContextCompat
import androidx.lifecycle.lifecycleScope
import com.example.tdm_android.constants.Constants
import com.example.tdm_android.functions.messageShort
import com.example.tdm_android.managers.UserManager
import com.example.tdm_android.models.User
import java.lang.Exception
import com.example.tdm_android.functions.restApiYesNoConsumptionLogin
import com.example.tdm_android.notifications.ServiceNotifications

class LoginActivity : AppCompatActivity() {

    private lateinit var tvCreateUser: TextView
    private lateinit var etUsername: EditText
    private lateinit var etPassword: EditText
    private lateinit var btnLogin: Button
    private lateinit var checkRememberUser: CheckBox
    private lateinit var imagePrincipal : ImageView

    private lateinit var pref: SharedPreferences
    private lateinit var editor: SharedPreferences.Editor

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val strName = intent.getStringExtra(Constants.STR_KEY_USERNAME)
        if (strName != null) { //Si se acaba de registrar
            messageShort(Constants.REGISTERED_SUCCESSFULLY_MESSAGE + strName)
        }

        initializeVariables()

        if ( pref.getBoolean(Constants.STR_CHECK_REMEMBER_USER, false) ) { //Si había elegido recordar usuario
            redirectToFilterActivity()
        } else{
            val origin = intent.getStringExtra("origin")
            if (origin == null){ //Solo cuando ingresa
                restApiYesNoConsumptionLogin(lifecycleScope, imagePrincipal, true)
            }
        }
        tvCreateUser.setOnClickListener { redirectToCreateUser() }
        btnLogin.setOnClickListener { login() }
    }

    private fun redirectToCreateUser() {
        val intent = Intent(this, RegisterActivity::class.java)
        startActivity(intent)
    }

    private fun login() {
        val strUsername = etUsername.text.toString()
        val strPassword = etPassword.text.toString()
        val dialog = Dialog(this@LoginActivity)
        dialog.setContentView(R.layout.custom_dialog)
        dialog.setCancelable(false)
        val tvDialogInfo = dialog.findViewById<View>(R.id.tvDialogInfo) as TextView
        val btnDialogOk = dialog.findViewById<View>(R.id.btnDialogOk) as Button
        btnDialogOk.setOnClickListener { dialog.dismiss() }
        if (strUsername.isEmpty() || strPassword.isEmpty()) {
            tvDialogInfo.text = Constants.COMPLETE_ALL_FIELD_MESSAGE
            dialog.show()
        } else if (isTheCredentialsAreValid(strUsername, strPassword)) {
            saveDataUser(checkRememberUser.isChecked)
            redirectToFilterActivity()
        } else {
            tvDialogInfo.text = Constants.INVALID_CREDENTIALS_MESSAGE
            dialog.show()
        }
    }

    private fun redirectToFilterActivity() {
        val intent = Intent(this, FilterActivity::class.java)
        intent.putExtra(Constants.STR_KEY_USERNAME, etUsername.text.toString())
        startActivity(intent)
    }

    private fun isTheCredentialsAreValid(strUsername: String?, strPassword: String): Boolean {
        try {
            val user: User? = UserManager.getInstance(this@LoginActivity)!!
                .getOneUserByField("username", strUsername)
            if (user != null && user.password == strPassword) {
                return true
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return false
    }

    private fun initializeVariables() {
        tvCreateUser = findViewById(R.id.createUser)
        btnLogin = findViewById(R.id.login)
        etUsername = findViewById(R.id.txtNameUser)
        etPassword = findViewById(R.id.txtPassword)
        checkRememberUser = findViewById(R.id.checkRememberUser)
        pref = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        editor = pref.edit()
        imagePrincipal = findViewById(R.id.image_principal)
    }

    private fun saveDataUser(isRememberUser: Boolean) {
        editor.putString(Constants.STR_USERNAME, etUsername.text.toString())
        editor.putString(Constants.STR_PASSWORD, etPassword.text.toString())
        editor.putBoolean(Constants.STR_CHECK_REMEMBER_USER, isRememberUser)
        editor.apply()
        createNotification()
    }

    private fun createNotification(){
        val intent = Intent(this, ServiceNotifications::class.java)
        ContextCompat.startForegroundService(this, intent)
    }

}